src/Symfony/Component/Mime/FileinfoMimeTypeGuesser.php

```diff
@@ -46,7 +48,7 @@ public function guessMimeType(string $path): ?string
             throw new LogicException(\sprintf('The "%s" guesser is not supported.', __CLASS__));
         }
 
-        if (false === $finfo = new \finfo(\FILEINFO_MIME_TYPE, $this->magicFile)) {
+        if (false === $finfo = self::$finfoCache[$this->magicFile] ??= new \finfo(\FILEINFO_MIME_TYPE, $this->magicFile)) {
```

Comment:
should we actually cache a `false` value ? This would break cases where the magic file gets created an another usage is done in the same process (especially given that we don't have an API to clear that cache)

Reply:
I don't think this use case is valid.

Reply:
Actually, this code dates back from PHP 5. An exception is thrown now, see https://3v4l.org/OpqUt
I've restored the previous behavior with a try/catch, but maybe we shouldn't ?